lib/diff_file.php

```diff
@@ -319,6 +319,7 @@ private function addEnsureForeignKey(rex_sql_foreign_key $foreignKey): string
         /** @var array<rex_sql_foreign_key::*, string> $modes */
         static $modes = [
             rex_sql_foreign_key::RESTRICT => 'rex_sql_foreign_key::RESTRICT',
+            /* rex_sql_foreign_key::NO_ACTION */ 'NO ACTION' => '',
```

Comment:
```suggestion
            /* rex_sql_foreign_key::NO_ACTION */ 'NO ACTION' => "'NO ACTION'",
```

Müsste es hier noch so heißen? Also damit in der Migrationsdatei dann auch `'NO ACTION'` als Parameter landet (dort halt auch der direkte Wert, da Konstante ggf. noch nicht vorhanden).

Reply:
Mir ist/war nicht klar, ob da wo die Ausgabe passiert, dann ein String anstelle eines Konstanten-Namens funktioniert...
Denn es ist ja `'rex_sql_foreign_key::RESTRICT' `und nicht `rex_sql_foreign_key::RESTRICT`
Und da NO ACTION ja der default Action Type ist, dachte ich es so gelöst zu haben.

Reply:
Es ist PHP-Code, der in die erzeugte Migrationsdatei geschrieben wird. Daher die doppelten Anführungszeichen in meinem Vorschlag. Also ein String mit der PHP-Expression `'NO ACTION'`.
Schau mal paar Zeilen tiefer in das Return-Statement. Aktuell erzeugst du dort invaliden PHP-Code. Mit deinem leeren String würde da sowas rauskommen:
```
    ->ensureForeignKey(new rex_sql_foreign_key('col', 'table', ['foo_id' => 'id'], , ))
```

Wir wollen bei NO ACTION aber das herausbekommen als PHP-Code:

```
    ->ensureForeignKey(new rex_sql_foreign_key('col', 'table', ['foo_id' => 'id'], 'NO ACTION', 'NO ACTION'))
```

Reply:
Bei den anderen Konstanten kommt sowas raus:

```
    ->ensureForeignKey(new rex_sql_foreign_key('col', 'table', ['foo_id' => 'id'], rex_sql_foreign_key::SET_NULL, rex_sql_foreign_key::SET_NULL))
```

Reply:
Wie immer: klar, wenn man es erklärt bekommt :-)